modules/tinymce/src/core/test/ts/browser/html/DomParserTest.ts

```diff
@@ -1416,5 +1416,35 @@ describe('browser.tinymce.core.html.DomParserTest', () => {
 
       assert.equal(serializedHtml, html);
     });
+
+    context('transparent elements', () => {
+      const getTransparentElements = (schema: Schema) => Arr.unique(Arr.map(Obj.keys(schema.getTransparentElements()), (s) => s.toLowerCase()));
+
+      it('TINY-9172: inline transparents should not get data-mce-block attribute', () => {
+        const parser = DomParser();
+        const html = '<p>' + Arr.map(getTransparentElements(parser.schema), (name) => `<${name}>text</${name}>`).join('') + '</p>';
```

Comment:
It might be more readable to break this up into 2 variables and use a template string maybe?